apps/crn-server/src/routes/research-outputs.route.ts

```diff
@@ -45,44 +47,61 @@ export const researchOutputRouteFactory = (
   );
 
   researchOutputRoutes.post('/research-outputs', async (req, res) => {
-    const { body, loggedInUser } = req;
+    const { body, loggedInUser, query } = req;
     const createRequest = validateResearchOutputPostRequestParameters(body);
     validateResearchOutputPostRequestParametersIdentifiers(createRequest);
 
-    if (
-      !loggedInUser ||
-      !hasCreateUpdateResearchOutputPermissions(
-        loggedInUser,
-        createRequest.teams,
-      )
-    ) {
+    const permissions = getUserPermissions(
+      loggedInUser as UserResponse,
+      'teams',
+      createRequest.teams,
+    );
+
+    const options = validateResearchOutputRequestQueryParameters(query);
+    const published = options.published ?? true;
+
+    const isRequestAllowed =
```

Comment:
I would say this is part of checking if the user has the permissions. I find this much more difficult to read now and I think there is no such separation of concerns as before. 

Maybe you should consider sending the `publish` query parameter across to the `hasCreateUpdateResearchOutputPermissions` function and keep the code more similar to what i was before? 

Before it was: `if i dont have permissions then throw` 
Now it is: `do i have permissions. But also is this a draft save or normal. And then do i have permissions if its one or the other, then throw`

Do you see what i mean?

Reply:
> Before it was: if i dont have permissions then throw

But before we didn't have to consider if the research output was published or not. So the permissions were only referring to user asap role/user team role

Not every condition in this Draft Feature depends on research output status (published/draft). For example, this Share Output Button does not depend on the research output status
![Screenshot 2023-03-01 at 08 38 06](https://user-images.githubusercontent.com/16595804/222128853-94c2fd2d-ef63-41af-ab28-a15d95462d12.png)

that's why I believe it's better to separate the permission referring to user asap role/user team role from the one of published or not